raiden/network/proxies/channel_manager.py

```diff
@@ -96,14 +99,51 @@ def new_netting_channel(self, other_peer: Address, settle_timeout: int) -> Addre
         if local_address == other_peer:
             raise SamePeerAddress('The other peer must not have the same address as the client.')
 
-        transaction_hash = estimate_and_transact(
-            self.proxy,
-            'newChannel',
+        # Check to see if a channel with this partner already exists
+        existing_channel = self.proxy.call(
+            'getChannelWith',
             other_peer,
-            settle_timeout,
         )
 
-        self.client.poll(unhexlify(transaction_hash), timeout=self.poll_timeout)
+        if existing_channel:
+            raise DuplicatedChannelError("Channel with given partner address already exists")
+
+        # Prevent concurrent attempts to open a channel with the same token and
+        # partner address.
+        if other_peer not in self.open_channel_transactions:
+            new_open_channel_lock = gevent.lock.Semaphore()
+
+            # Map from peer address to (lock, transaction_hash)
+            # TODO: Implement cleanup of keys in this dictionary based on reference-counting
+            self.open_channel_transactions[other_peer] = (new_open_channel_lock, None)
+
+        open_channel_lock = self.open_channel_transactions[other_peer][0]
+
+        if not open_channel_lock.locked():
+            with open_channel_lock:
+                transaction_hash = estimate_and_transact(
+                    self.proxy,
+                    'newChannel',
+                    other_peer,
+                    settle_timeout,
+                )
+                
+                self.open_channel_transactions[other_peer] = (new_open_channel_lock, transaction_hash)
+                # TODO: right now other threads will not see the timeout error if this times out.
+                # If we want those threads to see this timeout, we can have all blocking threads
+                # poll in the "else" case instead of blocking on the lock. Another option is to
+                # catch the exception here and mark the timeout in the open_channel_transactions map
+                # for all blocking threads to see.
+                self.client.poll(unhexlify(transaction_hash), timeout=self.poll_timeout)
+        
+        else:
+            open_channel_lock.wait()
+
+        # Read the result of the single "open channel" transaction across all threads
+        transaction_hash = self.open_channel_transactions[other_peer][1]
+        
+        if not transaction_hash:
+            raise RuntimeError('open channel transaction failed')
```

Comment:
move this inside the lock protected region, and use `async_result.set_exception` before raising the exception